components/brave_vpn/brave_vpn_service_desktop.cc

```diff
@@ -82,10 +87,13 @@ std::string GetBraveVPNPaymentsEnv() {
 
 BraveVpnServiceDesktop::BraveVpnServiceDesktop(
     scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
-    PrefService* prefs)
-    : BraveVpnService(url_loader_factory), prefs_(prefs) {
+    PrefService* prefs,
+    skus::SkusService* skus_service)
+    : BraveVpnService(url_loader_factory),
+      prefs_(prefs),
+      skus_service_(skus_service) {
   DCHECK(brave_vpn::IsBraveVPNEnabled());
-  DETACH_FROM_SEQUENCE(sequence_checker_);
+  DCHECK(skus_service_);
```

Comment:
you need to check for null here because the skus factory is only available to regular profiles and vpn service is available to all profiles

Reply:
well, I guess it will always load skus service from the regular profile since it's redirected in incognito, but that seems inconsistent with skus only being available for regular profiles in the first place. Why isn't skus redirected in incognito?

Reply:
I guess this could possibly make sense as-is because we don't want the skus sdk exposed in JS for incognito, but checking with @evq on this

Reply:
so I guess this is ok for now